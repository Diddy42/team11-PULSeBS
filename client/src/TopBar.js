import React from 'react';
import Navbar from 'react-bootstrap/Navbar'
import Button from 'react-bootstrap/Button'
import {Link} from 'react-router-dom';

class TopBar extends React.Component{
    render(){
        return <Navbar bg="dark" variant="dark">
                   <Navbar.Brand>PULSeBS</Navbar.Brand>
                   <Button variant="outline-light">Logout</Button>
               </Navbar>
    }
}



export default TopBar