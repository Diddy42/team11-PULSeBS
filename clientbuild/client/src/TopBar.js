import React from 'react';
import Navbar from 'react-bootstrap/Navbar'
import { Link, useHistory } from "react-router-dom";
import TeacherNavigationButton from './TeacherNavigationButton';
import BookingManagerNavButton from './BookingManagerNavButton';

class TopBar extends React.Component {

    render() {
        return <Navbar bg="dark" variant="dark">
            <HomeLink role={this.props.role}></HomeLink>
            {this.props.loggedin === false ?
                null
                :
                <>
                    <Navbar.Text className="text-align-center" data-testid="welcome-message">
                        {(this.props.role !== "support-officer") ? "Welcome " + this.props.userName : ""}
                    </Navbar.Text>

                    <ul className="ml-auto">

                        {
                            this.props.role === 'teacher' &&
                            <TeacherNavigationButton />
                        }

                        {
                            this.props.role === 'booking-manager' &&
                            <BookingManagerNavButton />
                        }

                        <Link className="text-light" to="/" onClick={() => this.props.logout()} data-testid="logout-link">
                            {"Logout "}
                            <svg className="bi bi-box-arrow-right" width="22" height="22" viewBox="0 0 16 16" fill="currentColor" xmlns="http://www.w3.org/2000/svg">
                                <path fillRule="evenodd" d="M11.646 11.354a.5.5 0 0 1 0-.708L14.293 8l-2.647-2.646a.5.5 0 0 1 .708-.708l3 3a.5.5 0 0 1 0 .708l-3 3a.5.5 0 0 1-.708 0z" />
                                <path fillRule="evenodd" d="M4.5 8a.5.5 0 0 1 .5-.5h9a.5.5 0 0 1 0 1H5a.5.5 0 0 1-.5-.5z" />
                                <path fillRule="evenodd" d="M2 13.5A1.5 1.5 0 0 1 .5 12V4A1.5 1.5 0 0 1 2 2.5h7A1.5 1.5 0 0 1 10.5 4v1.5a.5.5 0 0 1-1 0V4a.5.5 0 0 0-.5-.5H2a.5.5 0 0 0-.5.5v8a.5.5 0 0 0 .5.5h7a.5.5 0 0 0 .5-.5v-1.5a.5.5 0 0 1 1 0V12A1.5 1.5 0 0 1 9 13.5H2z" />
                            </svg>
                        </Link>

                    </ul>
                </>
            }
        </Navbar>
    }
}


const HomeLink = (props) => {
    const history = useHistory();
    const role = props.role;
    const supportOfficerHome = '/support-officer';

    const onHomeClick = () => {
        switch (role) {
            case 'support-officer':
                if (history.location.pathname !== supportOfficerHome) {
                    history.push(supportOfficerHome);
                }
                break;
            case 'teacher':
                break;
            default:
                break;
        }
    }

    return <Navbar.Brand
                data-testid="home-link"
                onClick={(ev) => { ev.preventDefault(); onHomeClick() }}
                style={{ cursor:'pointer' }}
            >
                PULSeBS
            </Navbar.Brand>
}

export default TopBar
