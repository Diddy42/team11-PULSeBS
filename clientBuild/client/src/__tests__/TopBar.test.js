import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import TopBar from '../TopBar';
import {BrowserRouter as Router} from 'react-router-dom';

describe("Logout render", () => {
    test("Logout render with user logged", () => {
        const loggedin = false;
        const {queryByTestId} = render(<Router><TopBar loggedin = {loggedin}/></Router>);
        const logoutLink = queryByTestId("logout-link");
        const welcomeMessage = queryByTestId("welcome-message");
        expect(logoutLink).not.toBeInTheDocument();
        expect(welcomeMessage).not.toBeInTheDocument();
    });

    test("Logout render without user logged", () => {
        const loggedin = true;
        const {queryByTestId} = render(<Router><TopBar loggedin = {loggedin}/></Router>);
        const logoutLink = queryByTestId("logout-link");
        const welcomeMessage = queryByTestId("welcome-message");
        expect(logoutLink).toBeInTheDocument();
        expect(welcomeMessage).toBeInTheDocument();
    });
});

test("Logout button", () => {
    const logout = jest.fn();
    const {queryByTestId} = render(<Router><TopBar logout = {logout}/></Router>);
    //const logoutLink = queryByTestId("logout-link");
    fireEvent.click(queryByTestId("logout-link"));
    expect(logout).toHaveBeenCalledTimes(1);
})
