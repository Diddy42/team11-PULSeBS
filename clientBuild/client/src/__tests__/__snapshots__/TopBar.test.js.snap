// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar renderer Whit User logged in 1`] = `
<nav
  className="navbar navbar-expand navbar-dark bg-dark"
>
  <span
    className="navbar-brand"
  >
    PULSeBS
  </span>
  <span
    className="ml-auto navbar-text"
    data-testid="welcome-message"
  >
    Welcome 
  </span>
  <a
    className="ml-auto"
    data-testid="logout-link"
    href="/"
    onClick={[Function]}
  >
    Logout 
    <svg
      className="bi bi-box-arrow-right"
      fill="currentColor"
      height="22"
      viewBox="0 0 16 16"
      width="22"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.646 11.354a.5.5 0 0 1 0-.708L14.293 8l-2.647-2.646a.5.5 0 0 1 .708-.708l3 3a.5.5 0 0 1 0 .708l-3 3a.5.5 0 0 1-.708 0z"
        fillRule="evenodd"
      />
      <path
        d="M4.5 8a.5.5 0 0 1 .5-.5h9a.5.5 0 0 1 0 1H5a.5.5 0 0 1-.5-.5z"
        fillRule="evenodd"
      />
      <path
        d="M2 13.5A1.5 1.5 0 0 1 .5 12V4A1.5 1.5 0 0 1 2 2.5h7A1.5 1.5 0 0 1 10.5 4v1.5a.5.5 0 0 1-1 0V4a.5.5 0 0 0-.5-.5H2a.5.5 0 0 0-.5.5v8a.5.5 0 0 0 .5.5h7a.5.5 0 0 0 .5-.5v-1.5a.5.5 0 0 1 1 0V12A1.5 1.5 0 0 1 9 13.5H2z"
        fillRule="evenodd"
      />
    </svg>
  </a>
</nav>
`;

exports[`TopBar renderer Without user logged in 1`] = `
<nav
  className="navbar navbar-expand navbar-dark bg-dark"
>
  <span
    className="navbar-brand"
  >
    PULSeBS
  </span>
</nav>
`;
